spec/models/user_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  context '#create' do
+    it 'is valid with the existing name' do
+      expect(User.create(name: 'Lily')).to be_valid
+    end
+
+    it 'is not valid with a blank name' do
+      expect(User.create(name: nil)).to_not be_valid
+    end
+
+    it 'is not valid with non-numeric posts_counter' do
+      expect(User.create(name: 'Lily', posts_counter: 'five')).to_not be_valid
+    end
+
+    it 'is not valid with float posts_counter' do
+      expect(User.create(name: 'Lily', posts_counter: 1.5)).to_not be_valid
+    end
+
+    it 'is not valid with negative posts_counter' do
+      expect(User.create(name: 'Lily', posts_counter: -1)).to_not be_valid
+    end
+
+    it 'is valid with integer posts_counter' do
+      expect(User.create(name: 'Lily', posts_counter: 5)).to be_valid
+    end
+  end
+
+  context '#three_most_recent_posts' do
+    before :all do
+      @user = User.create(name: 'Lily')
+      5.times { |post_i| Post.create(author: @user, title: (post_i + 1).to_s) }
+    end
+
+    it 'returns three most recent posts' do
+      expect(@user.three_most_recent_posts.length).to eq 3
+    end
+
+    it 'returns the most recent posts with titles 3, 4, 5' do
+      titles = []
+      @user.three_most_recent_posts.each do |post|
+        titles.push(post.title.to_i)
+      end
+      expect(titles).to all(be_between(3, 5))
```

Comment:
- You have done great so far :+1: :+1: But there are still some requirements you need to meet before moving to the next milestone. :+1: :100: 
- [ ] Looking at the screenshot below, you can see that you still have `11 test cases failing` I'll suggest you fix these. :+1:  

- - Screenshot reference:
 
![Screenshot from 2023-11-03 10-06-21](https://github.com/SaveryIV/blog-app-on-Rails/assets/107632780/6f5e5512-de73-4856-87e0-1b5263e9c2e0)

## From your communication with the previous reviewer, you mentioned getting an error of `Pending Migration` even after running `rails db:migrate` As I have explained to you over the Zoom call, running `rails db:migrate` only migrates the development database and the test files makes use of the test database, so you need to run `rails db:migrate RAILS_ENV=test` to migrate the test database. I know I have helped you correct this but do take note of it in your subsequent projects. :smile: :small_airplane: